interp/test.go

```diff
@@ -95,40 +95,43 @@ func (r *Runner) binTest(op syntax.BinTestOperator, x, y string) bool {
 	}
 }
 
-func stat(name string) os.FileInfo {
+func (r *Runner) stat(name string) os.FileInfo {
+	if !filepath.IsAbs(name) {
+		name = filepath.Join(r.Dir, name)
+	}
 	info, _ := os.Stat(name)
 	return info
 }
 
-func statMode(name string, mode os.FileMode) bool {
-	info := stat(name)
+func (r *Runner) statMode(name string, mode os.FileMode) bool {
+	info := r.stat(name)
 	return info != nil && info.Mode()&mode != 0
 }
 
 func (r *Runner) unTest(op syntax.UnTestOperator, x string) bool {
 	switch op {
 	case syntax.TsExists:
-		return stat(x) != nil
+		return r.stat(x) != nil
 	case syntax.TsRegFile:
-		info := stat(x)
+		info := r.stat(x)
 		return info != nil && info.Mode().IsRegular()
 	case syntax.TsDirect:
-		return statMode(x, os.ModeDir)
+		return r.statMode(x, os.ModeDir)
 	//case syntax.TsCharSp:
 	//case syntax.TsBlckSp:
 	case syntax.TsNmPipe:
-		return statMode(x, os.ModeNamedPipe)
+		return r.statMode(x, os.ModeNamedPipe)
 	case syntax.TsSocket:
-		return statMode(x, os.ModeSocket)
+		return r.statMode(x, os.ModeSocket)
 	case syntax.TsSmbLink:
 		info, _ := os.Lstat(x)
```

Comment:
@mvdan I missed those. I abstracted it in a function to reduce code duplication.